hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBDefinitionFactory.java

```diff
@@ -40,10 +46,24 @@ private DBDefinitionFactory() {
     dbMap.put(new OMDBDefinition().getName(), new OMDBDefinition());
   }
 
-  public static DBDefinition getDefinition(String dbName){
+  public static DBDefinition getDefinition(String dbName, String type){
```

Comment:
I am not sure if we need type here. I think this can be simplified as:

```java
 public static DBDefinition getDefinition(String dbName, String type){
    if (dbMap.containsKey(dbName)){
      return dbMap.get(dbName);
  }
    if (dbName.equals(RECON_SCM_DB_NAME)) {
      return new ReconSCMDBDefinition();
    } else if (dbName.startsWith(RECON_CONTAINER_KEY_DB)) {
      return new ReconDBDefinition(dbName);
    } else if (dbName.startsWith(RECON_OM_SNAPSHOT_DB)) {
      return new OMDBDefinition();
    }
    return null;
  }
```

Seems to be a good logic which just works.

Or actually `RECON_SCM_DB_NAME` also can be added to the map, so you don't need the firts condition

Reply:
You are right. At the start of writing this API, I was convinced I need a 'type' parameter. At the end of my patch, that was no longer true :) Forgot to remove this assumption. I will fix it.